src/Test/UnitTests/constraints.jl

```diff
@@ -132,3 +132,119 @@ function solve_affine_interval(model::MOI.ModelLike, config::TestConfig)
     end
 end
 unittests["solve_affine_interval"] = solve_affine_interval
+
+"""
+    solve_qcp_edge_cases(model::MOI.ModelLike, config::TestConfig)
+
+Test various edge cases relating to quadratically constrainted programs (i.e.,
+with a ScalarQuadraticFunction-in-Set constraint.
+
+If `config.solve=true` confirm that it solves correctly.
+"""
+function solve_qcp_edge_cases(model::MOI.ModelLike, config::TestConfig)
+    @testset "Duplicate on-diagonal" begin
+        # max x + 2y | y + x^2 + x^2 <= 1, x >= 0.5, y >= 0.5
+        MOI.empty!(model)
+        x = MOI.addvariables!(model, 2)
+        MOI.set!(model, MOI.ObjectiveSense(), MOI.MaxSense)
+        MOI.set!(model,
+            MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+            MOI.ScalarAffineFunction{Float64}(
```

Comment:
You don't have to explicitely specify the type `{Float64}`. At least it is inconsistent with other tests